calico_kubernetes/calico_kubernetes.py

```diff
@@ -269,75 +270,127 @@ def _get_api_token(self):
+                    tag = self._label_to_tag(key, value, namespace)
+                    args[label_ind + 1] = tag
+
+                # Remove empty strings and add to rule list
+                args = filter(None, args)
+                inbound_rules.append(args)
 
-        :return list() rules: the rules to be added to the Profile.
-        """
-        inbound_rules = [
-            {
-                'action': 'allow',
-            },
-        ]
-
-        outbound_rules = [
-            {
-                'action': 'allow',
-            },
-        ]
         return inbound_rules, outbound_rules
 
-    def _generate_profile_json(self, profile_name, rules):
+    def _apply_rules(self, profile_name, pod):
         """
-        Given a list of of Calico rules, generate a Calico Profile JSON blob
-        implementing those rules.
+        Generate a rules for a given profile based on annotations.
+        1) Remove Calicoctl default rules
+        2) Create new profiles based on annotations, and establish new defaults
 
-        :param profile_name: The name of the Calico profile
-        :type profile_name: string
-        :param rules: A tuple of (inbound, outbound) Calico rules
-        :type rules: tuple
-        :return: A JSON blob ready to be loaded by calicoctl
-        :rtype: str
-        """
-        inbound, outbound = rules
-        profile = {
-            'id': profile_name,
-            'inbound_rules': inbound,
-            'outbound_rules': outbound,
-        }
-        profile_json = json.dumps(profile, indent=2)
-        print('Final profile "%s": %s' % (profile_name, profile_json))
-        return profile_json
-
-    def _apply_rules(self, profile_name):
-        """
-        Generate a new profile with the default 'allow all' rules.
+        Exceptions:
+            If namespace = kube-system (internal kube services), allow all traffic
+            If no policy in annotations, allow from pod's Namespace
+            Outbound policy should allow all traffic
 
         :param profile_name: The profile to update
         :type profile_name: string
+        :param pod: pod info to parse
+        :type pod: dict()
         :return:
         """
         try:
             profile = self._datastore_client.get_profile(profile_name)
         except:
-            print("Error: Could not apply rules. Profile not found: %s, exiting" % profile_name)
+            print("ERROR: Could not apply rules. Profile not found: %s, exiting" % profile_name)
             sys.exit(1)
 
-        rules = self._generate_rules()
-        profile_json = self._generate_profile_json(profile_name, rules)
-        profile.rules = Rules.from_json(profile_json)
-        self._datastore_client.profile_update_rules(profile)
+        inbound_rules, outbound_rules = self._generate_rules(pod)
+
+        print "Removing Default Rules"
+
+        # TODO: This method is append-only, not profile replacement, we need to replace calicoctl calls
```

Comment:
If you do the bit below, then it would replace the profile (albeit in a slow hacky way) and we could remove this comment.